src/functions.date.js

```diff
@@ -108,28 +108,43 @@ for (let unit in s) {
 	}, {multiValued: true});
 }
 
-_.duration = $.extend(function($this, ms) {
+_.duration = $.extend(function($this, ms, terms = 1) {
 	if (arguments.length === 1) {
 		[ms, $this] = [$this, null];
 	}
-
-	var count = ms || 0;
-	var unit = "ms";
-
-	for (let nextUnit in s) {
-		var nextCount = _.msTo(nextUnit, ms);
-
-		if (nextCount === 0) {
-			break;
+	
+	let unitTime = ms || 0;
+	let timeLeft = ms || 0;
+	
+	if (ms === 0) {
+		terms = 1;
+	}
+	
+	let timeArray = [];
+	
+	while (timeArray.length < terms && (timeLeft > 0 || terms === 1)) {
```

Comment:
Why `terms === 1` as a separate condition? Wouldn't the rest suffice even in that case?

Reply:
As long as we have not filled up our array with the allotted # of terms, we will enter our while loop to add more terms if there is timeLeft to account for, or we had only 1 term asked for (in which case it's ok to return a zero value). So it allows for zero terms to be returned only in that zero term is the only term being returned. I'll add this in a comment to the code.